group/assignment6/parking_class.py

```diff
@@ -0,0 +1,54 @@
+class Parking:
+    """
+    Args:
+        constraints: {parking_lot: (permitted cars)}
+    """
+    def __init__(self, state):
+        self._state = state
+        self._inverted_state = self._compute_inverted_state()
+
+    def __len__(self):
+        return len(self._state)
+
+    def _compute_inverted_state(self):
+        inverted_state = list(range(len(self._state)))
+        for lot, car in enumerate(self._state):
+            inverted_state[car] = lot
+        return inverted_state
+
+    def get_car(self, lot):
+        return self._state[lot]
+
+    def get_lot(self, car):
+        """
+        :returns: int, number of a lot in which a given car is currently placed
+        """
+        return self._inverted_state[car]
+
+    def get_state(self):
```

Comment:
do you think that we need get method for it? maybe we can use public field state (I used properties in my code to resolve problem with input validation)

Reply:
i don't want to make the field 'state' public, this is exactly the reason why we need this class – to hide the implementation and maintain consistency between state and inverted_state. if a user of the class has access to the field 'state', they can change it and violate consistency between two lists.
is there more pythonic way to do that?

Reply:
@aliskin oh, I understand you, I don't know, I think that it your implementation is quite pythonic